pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	gitHubAPIURL = "https://api.github.com/repos/%v/%v?sort=created&direction=desc&per_page=1"
+)
+
+type GitHubResponse struct {
+	Number int `json:"number"`
+}
+
+type URLValidator struct {
+	Matched bool
+}
+
+// Match link with any one of provided validators.
+func (v Config) GetValidatorForURL(url string) URLValidator {
+	u := URLValidator{Matched: false}
+	for _, val := range v.Validators {
+		if val._regex.MatchString(url) {
+			if val.Type == "github" {
+				// Find rightmost index of match i.e, where regex match ends.
+				// This will be where issue/PR number starts. Split incase of section link and convert to int.
+				idx := val._regex.FindStringIndex(url)
+				stringNum := strings.Split(url[idx[1]:], "#")
+				num, err := strconv.Atoi(stringNum[0])
+				// If number in link does not exceed then link is valid. Otherwise will be checked by v.c.Visit.
+				if val._maxNum >= num && err == nil {
+					u.Matched = true
+					return u
+				}
+				return u
+			}
+			u.Matched = true
+			return u
+		}
+	}
+	return u
+}
+
+// If type is "github", change regex and add maxnum.
+func (v Config) validateGH() error {
+	for i := range v.Validators {
+		if v.Validators[i].Type == "github" {
+			regex, maxNum, err := getGitHubRegex(v.Validators[i].Regex)
+			if err != nil {
+				return err
+			}
+			v.Validators[i]._regex = regex
+			v.Validators[i]._maxNum = maxNum
+		}
+	}
+	return nil
+}
+
+// Get GitHub pulls/issues regex from repo name.
+func getGitHubRegex(repoRe string) (*regexp.Regexp, int, error) {
+	if repoRe != "" {
```

Comment:
ditto and please remember about this rule. If you invert this and improve readability a lot.

Also having one big IF at the beginning is a code smell. Why invoking this function in the first place if repoRe is empty? 🤔

Reply:
Sure! Will keep this in mind in future PRs.